ql/src/java/org/apache/hadoop/hive/ql/ddl/table/CreateTableOperation.java

```diff
@@ -140,6 +145,26 @@ private void createTableNonReplaceMode(Table tbl) throws HiveException {
       Table createdTable = context.getDb().getTable(tbl.getDbName(), tbl.getTableName());
       DataContainer dc = new DataContainer(createdTable.getTTable());
       context.getQueryState().getLineageState().setLineage(createdTable.getPath(), dc, createdTable.getCols());
+
+      // We did not create the table when moving the data files for a non-partitioned table and
+      // thus could not add a write notification required for a transactional table. Do that
+      // here, after we have created the table. Since this is a newly created table, listing all
+      // the files in the directory and listing only the ones corresponding to the given id
+      // doesn't have much difference.
+      if (!createdTable.isPartitioned() && AcidUtils.isTransactionalTable(createdTable)) {
+        org.apache.hadoop.hive.metastore.api.Table tTable = createdTable.getTTable();
+        Path tabLocation = new Path(tTable.getSd().getLocation());
+        List<Path> newFilesList = new ArrayList<>();
+        try {
+        context.getDb().listFilesInsideAcidDirectory(tabLocation,
```

Comment:
indentation issue ..line should be shifted right by 2 spaces

Reply:
Done.